### PR TITLE
use a local variable to calculate rtt

### PR DESCRIPTION
**Pull request by clmul**
This makes `Conn` safe for concurrent use if the message doesn't have a TSIG record.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -179,6 +177,7 @@ func (c *Client) exchange(m *Msg, a string) (r *Msg, rtt time.Duration, err erro
 	co.TsigSecret = c.TsigSecret
 	// write with the appropriate write timeout
 	co.SetWriteDeadline(time.Now().Add(c.getTimeoutForRequest(c.writeTimeout())))
+	t := time.Now()
```
Comment (tmthrgd) [inline client.go]: If you moved this up a line, we could avoid calling `time.Now` twice. It’s a small thing, but it does come with overhead.

**Comment by clmul** [comment]
> I don’t know how possible it would ever be to use Conn concurrently, this makes sense to me.

I reconsidered this and I agree that it's better to not use `Conn` concurrently.

File: client.go (hunk 1 above)
Comment (clmul) [reply client.go]: Fixed.

**Comment by miekg** [comment]
So `dns.Conn` came about because people wanted to use `laddr` and other `Dial` properties which were not exposed by just using Exchange. I think most of these tweaks should just move to `Client`, so we can drop all these exposed bits.

**Pull request merged**